### PR TITLE
1997 new object field step 1 create page

### LINKED ISSUE #1997: New object field - Step 1 - Create page
Warning: this issue is part of a series of issues aiming to create the "Field Creation flow". This is issue number 1.

## Context

See #1767 

For this issue:
- Create an empty page with the route: /settings/{:pluralObjectName}/new-field/step1
- Use the breadcrumb component from #1814.

## Figma

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11336-63097&mode=design&t=xSdrIMXJed2ZTg34-11

## Behavior

**GIVEN** I am on the Object Detail page
**AND** the object has disabled fields
**WHEN** I click on the "Add field" button
**THEN** I navigate to the New field - Step 1 page

**GIVEN** I am on an New Field - Step 1 page
**THEN** I see a breadcrumb

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #1997

### CODE REVIEW

File: front/src/modules/types/SettingsPath.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export enum SettingsPath {
   Objects = 'objects',
   ObjectDetail = 'objects/:pluralObjectName',
   ObjectEdit = 'objects/:pluralObjectName/edit',
+  ObjectNewFieldStep1 = 'objects/:pluralObjectName/new-field/step1',
```
Comment (charlesBochet) [inline front/src/modules/types/SettingsPath.ts:7]: step-1

File: front/src/pages/settings/data-model/SettingsObjectDetail.tsx [hunk 2]
```diff
@@ -103,6 +103,7 @@ export const SettingsObjectDetail = () => {
           title="Add Field"
           size="small"
           variant="secondary"
+          onClick={() => navigate(`./new-field/step1`)}
```
Comment (charlesBochet) [inline front/src/pages/settings/data-model/SettingsObjectDetail.tsx:106]: step-1

File: front/src/pages/settings/data-model/SettingsObjectNewField/SettingsObjectNewFieldStep1.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import { activeObjectItems } from '@/settings/data-model/constants/mockObjects';
+import { objectSettingsWidth } from '@/settings/data-model/constants/objectSettings';
+import { AppPath } from '@/types/AppPath';
+import { IconSettings } from '@/ui/display/icon';
+import { SubMenuTopBarContainer } from '@/ui/layout/page/SubMenuTopBarContainer';
+import { Breadcrumb } from '@/ui/navigation/bread-crumb/components/Breadcrumb';
+
+const StyledContainer = styled.div`
+  height: fit-content;
```
Comment (charlesBochet) [inline front/src/pages/settings/data-model/SettingsObjectNewField/SettingsObjectNewFieldStep1.tsx:13]: create a SettingsPageContainer that we can re-use in all Settings page. We can pass width as a prop

File: front/src/pages/settings/data-model/SettingsObjectDetail.tsx (hunk 2 above)
Comment (thaisguigon) [inline front/src/pages/settings/data-model/SettingsObjectDetail.tsx:106]: The Step 1 page is only accessible if there are disabled fields:

```suggestion
          onClick={() =>
            navigate(
              disabledFieldItems.length
                ? './new-field/step1'
                : './new-field/step2',
            )
          }
```

We can implement it now or when we create the Step 2 page if you want.

**Review by thaisguigon — APPROVED** [review]

**Review by charlesBochet — APPROVED** [review]
LGTM

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**